repo_resource/common.py

```diff
@@ -121,9 +198,16 @@ class Repo:
     such as init/sync and manifest
     """
 
-    def __init__(self, workdir=CACHEDIR):
+    def __init__(self, url, revision='HEAD', name='default.xml',
+                 depth=-1, workdir=CACHEDIR):
         self.__workdir = workdir
         self.__oldpwd = None
+        self.__url = url
```

Comment:
adding these parameters could probably be a seperate (preparatory) patch

Reply:
Done